src/main/groovy/val/Idator.groovy

```diff
@@ -22,6 +79,280 @@ class Idator {
      */
     //TODO: allow per instance extension
     void registerChecker(String name, Closure closure) {
-        DefinitionCheck.metaClass[name] = closure
+        extraChecks[name] = closure
+    }
+
+    def methodMissing(String name, args) {
+        if (extraChecks[name])
+            extraChecks[name](*args)
+        else throw new MissingMethodException(name, this.class, args)
     }
```

Comment:
caching of the methods further up the dispatch chain should be done later but is also low priority since the definition phase can be slow